static/app/components/modals/sudoModal.tsx

```diff
@@ -56,23 +81,37 @@ class SudoModal extends React.Component<Props, State> {
       return;
     }
 
-    this.setState({busy: true}, () => {
+    this.setState({busy: true, isFirstStep: true}, () => {
       retryRequest().then(() => {
         this.setState({busy: false}, closeModal);
       });
     });
   };
 
-  handleError = () => {
-    this.setState({busy: false, error: true});
+  handleError = err => {
+    let errType = '';
+    if (err.status === 403) {
+      errType = 'invalidPassword';
```

Comment:
rather than translating the error code to a string, then translating that string into a more descriptive string, can we just save the code?

Or maybe create and save an enum?